### PR TITLE
[#1431] Add Number validation to template plugins Pop, Word and Photo

### PR DESCRIPTION
**Pull request by mjschranz**
https://webmademovies.lighthouseapp.com/projects/65733/tickets/1431

### CODE REVIEW

File: templates/supported/shared/plugins/photo/popcorn.photo.js [hunk 1]
```diff
@@ -28,6 +28,18 @@
       } )
  *
  */
+
+ function normalize( value, minWidth, maxWidth ) {
```
Comment (cadecairos) [inline templates/supported/shared/plugins/photo/popcorn.photo.js]: this function is not used.

File: templates/supported/shared/plugins/photo/popcorn.photo.js [hunk 2]
```diff
@@ -94,7 +94,11 @@
       _setup: function( options ) {
         var img,
             target = document.getElementById( options.target ),
-            context = this;
+            context = this,
+            width = options.width + "px",
+            height = options.height + "px",
+            top = options.top + "px",
+            left = options.left + "px";
```
Comment (ScottDowne) [inline templates/supported/shared/plugins/photo/popcorn.photo.js]: Why are you caching this value if it is only used once?

File: templates/supported/shared/plugins/photo/popcorn.photo.js [hunk 3]
```diff
@@ -106,10 +110,10 @@
...
           options._container.style.position = "absolute";
           options._container.style.display = "none";
-          options._container.style.width = options.width;
-          options._container.style.height = options.height;
-          options._container.style.top = options.top;
-          options._container.style.left = options.left;
+          options._container.style.width = width;
+          options._container.style.height = height;
+          options._container.style.top = top;
+          options._container.style.left = left;
```
Comment (ScottDowne) [inline templates/supported/shared/plugins/photo/popcorn.photo.js]: Related to my above comment, why not just do:

options._container.style.left = options.left + "px";

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/bd90372fb19da0a/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/add24a8f7398cd7/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/bd90372fb19da0a/output.txt

Total script time: 0.95 mins

Server is running on http://pr452.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/add24a8f7398cd7/output.txt

Total script time: 0.24 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/3a0e098617f06ac/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/ff783742bc6ae16/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/3a0e098617f06ac/output.txt

Total script time: 0.92 mins

Server is running on http://pr452.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/ff783742bc6ae16/output.txt

Total script time: 0.32 mins

File: templates/supported/newscaster/butter/template.js [hunk 4]
```diff
@@ -57,6 +57,7 @@ document.addEventListener( "DOMContentLoaded", function( e ){
             if ( trackEvent.type === "photo" ) {
               // Prevent default draggable behaviour of images
               trackEvent.popcornTrackEvent._image.addEventListener( "mousedown", function(e) {
+                console.log("derP");
```
Comment (ScottDowne) [inline templates/supported/newscaster/butter/template.js]: Nuke the derp.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/b541754ede881a6/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/cbe66aef256fa61/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/b541754ede881a6/output.txt

Total script time: 1.23 mins

Server is running on http://pr452.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/cbe66aef256fa61/output.txt

Total script time: 0.39 mins

**Pull request merged**